packages/cli/src/cmds/navie.ts

```diff
@@ -0,0 +1,202 @@
+
+  const useLocalNavie = () => {
+    if (argv.navieProvider === 'local') {
+      warn(`Using local Navie provider due to explicit --navie-provider=local option`);
+      return true;
+    }
+
+    if (argv.navieProvider === 'remote') {
+      warn(`Using remote Navie provider due to explicit --navie-provider=remote option`);
+      return false;
+    }
+
+    const aiEnvVar = Object.keys(process.env).find((key) => AI_KEY_ENV_VARS.includes(key));
+    if (aiEnvVar) {
+      warn(`Using local Navie provider due to presence of environment variable ${aiEnvVar}`);
+      return true;
+    }
+
+    warn(
+      `--navie-provider option not provided, and none of ${AI_KEY_ENV_VARS.join(
+        ' '
+      )} are available. Using remote Navie provider.`
+    );
+    return false;
+  };
+
+  const applyAIOptions = (navie: LocalNavie | RemoteNavie) => {
+    if (aiOptions) {
+      for (const option of aiOptions) {
+        const [key, value] = option.split('=');
+        if (key && value) {
+          navie.setOption(key, value);
+        }
+      }
+    }
+    if (agentMode) {
+      navie.setOption('explainMode', agentMode);
+    }
+  };
+
+  const buildLocalNavie = (
+    threadId: string | undefined,
+    contextProvider: ContextV2.ContextProvider,
+    projectInfoProvider: ProjectInfo.ProjectInfoProvider,
+    helpProvider: Help.HelpProvider
+  ) => {
+    const navie = new LocalNavie(threadId, contextProvider, projectInfoProvider, helpProvider);
+    applyAIOptions(navie);
+
+    let START: number | undefined;
+    const logEvent = (event: InteractionEvent) => {
+      if (!logNavie) return;
+
+      if (!START) START = Date.now();
+
+      const elapsed = Date.now() - START;
+      process.stderr.write(chalk.gray(`${elapsed}ms `));
+      process.stderr.write(chalk.gray(event.message));
+      process.stderr.write(chalk.gray('\n'));
+    };
+
+    navie.on('event', logEvent);
+    return navie;
+  };
+
+  const buildRemoteNavie = (
+    threadId: string | undefined,
+    contextProvider: ContextV2.ContextProvider,
+    projectInfoProvider: ProjectInfo.ProjectInfoProvider,
+    helpProvider: Help.HelpProvider
+  ) => {
+    loadConfiguration(true);
+    const navie = new RemoteNavie(threadId, contextProvider, projectInfoProvider, helpProvider);
+    applyAIOptions(navie);
+    return navie;
+  };
+
+  return useLocalNavie() ? buildLocalNavie : buildRemoteNavie;
+}
+
+function attachNavie(navie: INavie) {
+  return navie
+    .on('error', (err) => {
+      warn(err);
+      process.exitCode = 1;
+    })
+    .on('token', (token) => process.stdout.write(token));
+}
+
+interface NavieCmdArgs extends NavieCommonCmdArgs {
+  questionPath: string;
+}
+
+export const command = 'navie';
+export const describe = 'Explain a question using Navie';
+
+export function builder<T>(args: yargs.Argv<T>): yargs.Argv<T & NavieCommonCmdArgs> {
+  return commonNavieArgsBuilder(args).positional('question-path', {
```

Comment:
For people who are not as shell-adept (and for Windows users) I'd recommend that the default positional argument be the question text. And that question input from a file be a separate option.